src/FormContainer.tsx

```diff
@@ -13,7 +14,8 @@ const makeWrapper = <T extends {}>(config: IFormConfig<T>) => (WrappedComponent:
             this.state = {
                 model: config.initialModel || {},
                 touched: {},
-                inputs: {}
+                inputs: {},
+                shouldValidate: {}
```

Comment:
Could we get away without using a wrapper component and just populate `shouldValidate` as each bound input is rendered?

Reply:
what would be the best way to do it?

Reply:
I was thinking we could have a side effect when invoking `bindInput` that could update the state to register the field. Then the validation would only occur on registered fields.

That would work, but it wouldn't be able to unregister the field on unmount so that's where I draw a blank.

I like the thinking behind the approach here, but it feels like too much boilerplate. I'm trying to think if there's another way. 🤔

Reply:
Is HOC the boilerplate you're talking about?
From my POV it's the only solution that gives us full control (e.g. we can pass ref down of not) and allows to stay declarative.

One other thing to consider: child input might have a locked down interface (e.g. with Typescript) and won't allow for any arbitrary props like `shouldValidate`

What's left is to move `shouldValidate` to validators, but it has less access to conditional logic in JSX that can be reused.

Reply:
same concept in redux-form btw:
https://codesandbox.io/s/8zwvq2zr

so seems like I'm not inventing the wheel here ;)